sandbox/modules/identity/src/main/java/org/opensearch/identity/configuration/ConfigurationRepository.java

```diff
@@ -313,4 +337,40 @@ private Map<CType, SecurityDynamicConfiguration<?>> validate(Map<CType, Security
 
         return conf;
     }
+
+    public void initializeExtensionsSecurity() {
+        SecurityDynamicConfiguration<ExtensionSecurityConfig> extensionsSecurityConfig = SecurityDynamicConfiguration.empty();
+        Map<String, DiscoveryExtensionNode> extensionIdMap = IdentityPlugin.GuiceHolder.getExtensionsManager().getExtensionIdMap();
+        if (extensionIdMap != null) {
+            for (String extensionId : extensionIdMap.keySet()) {
+                ExtensionSecurityConfig es = new ExtensionSecurityConfig();
+                es.setSigningKey(ExtensionsSecretsGenerator.generateSigningKey());
+                extensionsSecurityConfig.putCObject(extensionId, es);
+            }
+
+            String configType = CType.EXTENSIONSECURITY.toLCString();
+
+            try {
+                final IndexRequest indexRequest = new IndexRequest(this.identityIndex).id(configType)
+                    .opType(DocWriteRequest.OpType.INDEX)
```

Comment:
Why OpType INDEX here? It will replace any existing doc, is that the idea here?

Reply:
Its set as empty initially from lines 127 - 138 so this is INDEX so that it will update. `CREATE` is only used the very first time, all other config updates will be `INDEX`.